### PR TITLE
Valhalla Implementation

### PR DESCRIPTION
**Pull request by killjoy1221**
Also improves extensibility of `SkinServer`

Opening for discussion.

### CODE REVIEW

**Comment by Sollace** [comment]
Already looked through an earlier commit of this and I have no objections to including both of our server types.

**Comment by killjoy1221** [comment]
I might add an endpoint at the root directory that can be used to identify
the server.

**Comment by Sollace** [comment]
Sure. If you'd like we can standardise it and have both servers implement the same format to identify themselves.

File: src/main/java/com/minelittlepony/MineLittlePony.java [hunk 1]
```diff
@@ -57,7 +59,7 @@
         ms.registerMetadataSectionType(new PonyDataSerialzier(), IPonyData.class);
 
         // This also makes it the default gateway server.
```
Comment (killjoy1221) [inline src/main/java/com/minelittlepony/MineLittlePony.java]: This is actually no longer true.

File: src/hdskins/java/com/voxelmodpack/hdskins/skins/LegacySkinServer.java [hunk 2]
```diff
@@ -144,24 +142,11 @@ private static void verifyServerConnection(Session session, String serverId) thr
...
-        throw new IllegalArgumentException("server format string was not correct");
-    }
-
     @Override
     public String toString() {
-        return MoreObjects.toStringHelper(this)
-                .add("address", address)
-                .add("gateway", gateway)
-                .toString();
+        return new ToStringBuilder(this, IndentedToStringStyle.INSTANCE)
```
Comment (Sollace) [inline src/hdskins/java/com/voxelmodpack/hdskins/skins/LegacySkinServer.java]: Why not use @Expose and a Gson?

Comment (killjoy1221) [reply src/hdskins/java/com/voxelmodpack/hdskins/skins/LegacySkinServer.java]: This is for displaying on screen. It looks like this
```
LegacySkinServer
  address=XXX
  gateway=XXXX
```

**Pull request merged**